### DISCUSSION TITLE
Set the Prescribed Velocity using a file

### DISCUSSION THREAD

**Discussion by kuailebozai** [Anura3D/Anura3D_OpenSource#187] [Q&A]
Hello everyone！ I am trying to set the Prescribed Velocity using an external file ".PVF" and I have changed the name of the file under the $$PRESCRIBED_VELOCITY flag, but the calculations say that the file doesn't exist, I don't know what is wrong, please help me, thank you very much!
![微信图片_20240219071943](https://github.com/Anura3D/Anura3D_OpenSource/assets/52518762/b19c26f8-427d-409b-bd4a-c62d50a50fb3)
![微信图片_20240219072455](https://github.com/Anura3D/Anura3D_OpenSource/assets/52518762/967171f2-ffcd-47d4-9018-e42160215247)
![微信图片_20240219072038](https://github.com/Anura3D/Anura3D_OpenSource/assets/52518762/90b40f08-c271-4826-abd2-55025274253d)

**Comment by murphy-john** [comment]
The PVF file is a text file you create with the time in Column 1 and a velocity in Column 2. The details are explained in the ReadCalculationData.FOR file under the ReadPrescribedAccelerationOrVelocityFromFile() subroutine.

**Reply by kuailebozai** [reply]
Hello Murphy John, I set it up according to the instructions of the ReadPrescribedAccelerationOrVelocityFromFile() program. I created it PVF file, but the program cannot find this file during calculation. I am not sure if there is an error in my file settings. Have you ever made such settings before? I hope you can guide me to correct the error. Thank you very much for your reply.
Best wishes

**Reply by Mangwuha** [reply]
@kuailebozai 
Hello, I am a student who is learning Anura3D, may I ask whether the question you asked above has been solved? I was also interested in applying variable speeds, but didn't know how to do it properly.
Best wishes!

**Reply by CPLA81192** [reply]
When I use the prescribed velocity and set the moving mesh, when the mesh moves, the material points do not move, which causes the material points to jump out of the mesh. What is the reason for this and how can it be solved?
